api/apps/api/test/project-scenarios.e2e-spec.ts

```diff
@@ -126,59 +128,107 @@ describe('ScenariosModule (e2e)', () => {
     const response = await fixtures.WhenAcquiringLockForScenarioAsOwner();
-    fixtures.ThenScenarioLockInfoIsReturned(response);
+    fixtures.ThenScenarioLockInfoForOwnerIsReturned(response);
   });
 
-  it('Fails to acquire a lock for a scenario as there was one already', async () => {
+  it('Fails to acquire lock for a scenario as there was one already', async () => {
     await fixtures.GivenScenarioWasCreated();
     await fixtures.GivenContributorWasAddedToScenario();
 
     let response = await fixtures.WhenAcquiringLockForScenarioAsOwner();
-    fixtures.ThenScenarioLockInfoIsReturned(response);
+    fixtures.ThenScenarioLockInfoForOwnerIsReturned(response);
 
     response = await fixtures.WhenAcquiringLockForScenarioAsContributor();
     fixtures.ThenScenarioIsLockedIsReturned(response);
   });
 
-  it('Fails to update a scenario as there was a lock in place by a different user', async () => {
+  it('Fails to update scenario as there was a lock in place by a different user', async () => {
     await fixtures.GivenScenarioWasCreated();
     await fixtures.GivenContributorWasAddedToScenario();
 
     let response = await fixtures.WhenAcquiringLockForScenarioAsOwner();
-    fixtures.ThenScenarioLockInfoIsReturned(response);
+    fixtures.ThenScenarioLockInfoForOwnerIsReturned(response);
 
     response = await fixtures.WhenUpdatingScenarioAsContributor();
     fixtures.ThenScenarioIsLockedIsReturned(response);
   });
 
-  it('Updates a scenario correctly as lock is in place by same user', async () => {
+  it('Updates scenario correctly as lock is in place by same user', async () => {
     await fixtures.GivenScenarioWasCreated();
     await fixtures.GivenContributorWasAddedToScenario();
 
     let response = await fixtures.WhenAcquiringLockForScenarioAsOwner();
-    fixtures.ThenScenarioLockInfoIsReturned(response);
+    fixtures.ThenScenarioLockInfoForOwnerIsReturned(response);
 
     response = await fixtures.WhenUpdatingScenarioAsOwner();
     fixtures.ThenScenarioIsUpdated(response);
   });
 
+  it('Releases scenario lock correctly', async () => {
+    await fixtures.GivenScenarioWasCreated();
+    const userToken = await fixtures.GivenUserIsLoggedIn('owner');
+
+    let response = await fixtures.WhenAcquiringLockForScenarioAsOwner();
+    fixtures.ThenScenarioLockInfoForOwnerIsReturned(response);
+
+    response = await fixtures.WhenReleasingLockForScenario(userToken);
+    fixtures.ThenNoContentIsReturned(response);
+  });
+
+  it('Releases scenario lock correctly because user is not owner of lock but is project owner', async () => {
+    await fixtures.GivenScenarioWasCreated();
+    await fixtures.GivenContributorWasAddedToScenario();
+    const userToken = await fixtures.GivenUserIsLoggedIn('owner');
+
+    let response = await fixtures.WhenAcquiringLockForScenarioAsContributor();
+    fixtures.ThenScenarioLockInfoForContributorIsReturned(response);
+
+    response = await fixtures.WhenReleasingLockForScenario(userToken);
```

Comment:
renaming this may make the intent even clearer (I'd say it's ok as is because things can be easily figured out by lifting one's eyes by a few lines of code), though we'd have to also refactor the function to avoid passing in the `userToken` (or it'd be actually worse):
```suggestion
    response = await fixtures.WhenReleasingLockForScenarioAsOwner();
```

Reply:
same below

Reply:
Should it be enough if I rename the token as `ownerToken`? It's because I used a similar pattern on PR 815 for some other endpoints. I wanted to avoid what you pointed out in a previous PR of the repetition of having the same endpoint 3 times for each role inside an entity.